api/src/main/java/io/opentelemetry/api/trace/TraceStateBuilder.java

```diff
@@ -10,27 +10,31 @@
 import java.util.Objects;
 import javax.annotation.Nullable;
 
-/** A builder of {@link TraceState}. */
+/**
+ * A builder of {@link TraceState}. This implementation does full validation of the keys and values
```

Comment:
Thinking about your question about w3c prefix, it's a bit weird. The w3c'ness is only in this builder, not the TraceState, not sure how best to deal with that. I wonder if we should remove the validation from this class and do it in the propagator?

Reply:
yeah...it's a weird thing to be first-class in the APIs, and not constrained to be part of the propagator, TBH. I was pondering only making this class prefixed with W3C, and leaving the TraceState itself to be generic.

Reply:
I think I'd leave it as is then - the builder of TraceState being TraceStateBuilder seems clearer even with the w3c caveats.